### PR TITLE
Feature/EUI 7018 new sar

### PR DESCRIPTION
**Pull request by DavidJayakumar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-7018
https://tools.hmcts.net/jira/browse/EUI-7837

### Change description ###

Delete the secondary role associated with the SAR request 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/workAllocation/util.ts [hunk 1]
```diff
@@ -748,13 +757,26 @@ export function mapRoleCaseData(roleAssignment: RoleAssignment, caseDetail: Case
     reviewerRoleCategory: roleAssignment.attributes.reviewerRoleCategory,
   };
 }
+
+export function checkIsNew(roleAssignment: RoleAssignment, newRoleAssignmentList: RoleAssignment[]): boolean {
+  console.log('getStartDate(roleAssignment)', getStartDate(roleAssignment))
```
Comment (connorpgpmcelroy) [inline api/workAllocation/util.ts:762]: console.log should be removed

**Review by connorpgpmcelroy — DISMISSED** [review]
Approved with requested change

**Review by Surekha-HMCTS — APPROVED** [review]

File: src/work-allocation/components/work-field/work-field.component.html [hunk 2]
```diff
@@ -50,6 +50,6 @@
     <exui-derived-icon-field [workField]="workField" [sourceColumn]="config.sourceColumn" [matchValue]="config.matchValue"></exui-derived-icon-field>
   </ng-container>
   <ng-container *ngSwitchCase="fieldType.ACCESS_VIEW">
-    <exui-access-view-field [workField]="workField" [id]="workField.id"></exui-access-view-field>
+    <exui-access-view-field [workField]="workField" [id]="workField.id" (click)="clickSubject.next(workField)"></exui-access-view-field>
```
Comment (MateiHMCTS) [inline src/work-allocation/components/work-field/work-field.component.html:53]: Replace with the new onClick method.

**Review by MateiHMCTS — APPROVED** [review]

File: api/specificAccessOrchastrator/index.spec.ts [hunk 3]
```diff
@@ -4,21 +4,23 @@ import 'mocha';
 import * as sinon from 'sinon';
 import * as sinonChai from 'sinon-chai';
 import { mockReq, mockRes } from 'sinon-express-mock';
+import { getTaskType, orchestrationSpecificAccessRequest, postCreateTask, specificAccessRequestUpdateAttributes } from '.';
```
Comment (connorpgpmcelroy) [inline api/specificAccessOrchastrator/index.spec.ts:7]: Why has this line been moved from below? Should be the last line as it is imported from same folder

**Review by Surekha-HMCTS — APPROVED** [review]

File: api/specificAccessOrchastrator/index.ts [hunk 4]
```diff
@@ -194,36 +194,44 @@ export async function specificAccessRequestUpdateAttributes(req: EnhancedRequest
       attributes: {
         caseId: [caseId],
       },
-    }, { headers });
+    }, {headers});
 
-    const singleRoleAssignment = roleAssignmentQueryResponse.data.roleAssignmentResponse[0];
+    const singleRoleAssignment = roleAssignmentQueryResponse.data.roleAssignmentResponse
+      .find(r => r.roleName === 'specific-access-granted' ||  r.roleName === 'specific-access-denied');
```
Comment (connorpgpmcelroy) [inline api/specificAccessOrchastrator/index.ts:200]: Very minor comment, should use role instead of r here

File: api/workAllocation/util.ts [hunk 5]
```diff
@@ -520,14 +520,20 @@ export function filterMyAccessRoleAssignments(roleAssignmentList: RoleAssignment
...
+
   const specificRoleAssignments = filterMyAccessRoleAssignments(roleAssignmentList);
 
   const cases = await getCaseIdListFromRoles(specificRoleAssignments, req);
+  return mapCasesFromData(cases, specificRoleAssignments, newRoleAssignment);
+}
 
-  return mapCasesFromData(cases, specificRoleAssignments);
+export function getAccessGrantedRoleAssignments(roleAssignmentList: RoleAssignment[]): RoleAssignment[] {
+  return roleAssignmentList.filter(r => r.roleName === 'specific-access-granted')
```
Comment (connorpgpmcelroy) [inline api/workAllocation/util.ts:533]: again, would prefer use of role here

File: api/workAllocation/util.ts [hunk 6]
```diff
@@ -520,14 +520,20 @@ export function filterMyAccessRoleAssignments(roleAssignmentList: RoleAssignment
...
 }
 
-export function constructElasticSearchQuery(caseIds: any[], page: number, size: number): ElasticSearchQuery [] {
+export function constructElasticSearchQuery(caseIds: any[], page: number, size: number): ElasticSearchQuery[] {
```
Comment (connorpgpmcelroy) [inline api/workAllocation/util.ts:536]: caseIds is a list of string?

File: src/work-allocation/components/work-field/work-field.component.ts [hunk 7]
```diff
@@ -3,9 +3,9 @@ import { FieldConfig } from '../../models/common';
 import { Case } from '.././../models/cases';
 import { Task } from '.././../models/tasks';
 
-import { FieldType } from '../../enums';
 import { Subject } from 'rxjs';
 import { take } from 'rxjs/operators';
```
Comment (connorpgpmcelroy) [inline src/work-allocation/components/work-field/work-field.component.ts:7]: Imports of rxjs should be before ../.. imports

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]

File: api/workAllocation/util.ts (hunk 6 above)
Comment (DavidJayakumar) [reply api/workAllocation/util.ts:536]: The change is in the white space. Changing this has to be tested and not releated to this JIRA

**Pull request merged**